src/app/api/user/[uid]/edit-profile/route.ts

```diff
@@ -0,0 +1,112 @@
+import { NextRequest, NextResponse } from "next/server";
```

Comment:
nit: I think we don't need the `edit-profile` segment in API route. The reason is that `PATCH` specifies that it's an update operation, so it's implicitly a "edit-profile" operation.